### ISSUE TITLE
Beacon startup error: DB error when reading head state:HotColdDBError

### ISSUE THREAD

**Issue by acequicksilver** [sigp/lighthouse#1318]
## Description

When starting a beacon node after for the second time . The following error is observed:

beacon_node_1       | Jun 29 16:22:36.597 WARN Ethereum 2.0 is pre-release. This software is experimental.
beacon_node_1       | Jun 29 16:22:36.598 INFO Using default testnet                   default: altona-v3
beacon_node_1       | Jun 29 16:22:36.598 INFO Data directory initialised              datadir: /root/.lighthouse
beacon_node_1       | Jun 29 16:22:36.617 INFO Starting beacon chain                   method: resume, service: beacon
beacon_node_1       | Failed to start beacon node: DB error when reading head state: HotColdDBError(BlockReplayBlockError(HeaderInvalid { reason: ParentBlockRootMismatch { state: 0xae81b076b21c3b93c1b8c2c2c73f8c00102dc0d74af0f78405a0c74703be3250, block: 0xa41eee2698f87b389edcd1a4f951ec5138f8f940b8d4079d93ffd6504924721a } }))
lighthouse-docker_beacon_node_1 exited with code 1

## Version

Running from within a Linux Ubuntu docker environment

Built from master - 27th June

Rust version 1.43.1
Lighthouse 

## Present Behaviour

Beacon node cannot start

## Expected Behaviour

Beacon node should attempt to sync with network when started.

**Comment by michaelsproul** [comment]
I'm struggling to work out what might have happened to lead you to this case... it might indeed be an interesting bug. Neither of the block hashes `0xa41eee2698f87b389edcd1a4f951ec5138f8f940b8d4079d93ffd6504924721a` or `0xae81b076b21c3b93c1b8c2c2c73f8c00102dc0d74af0f78405a0c74703be3250` seem to be blocks on Altona. Is it possible that you had a datadir lying around from running a Witti node previously? Did you re-use the same `lighthouse-docker` directory (which would have contained `.lighthouse/beacon`, possibly with Witti state)?

You mention that this was the second time you ran the beacon node, when did you first run it?

**Comment by acequicksilver** [comment]
So these are the steps I took.

1. git cloned https://github.com/sigp/lighthouse.git
2. Built docker image from source(lighthouse:local)
3. Modified your Git docker-compose file to reference local lighthouse image and executed docker-compose up in the lighthouse-docker directory
 - Initially ran on Either Friday or Saturday 26th/27th June for the first time and stopped the docker container a few mins after
4. Executed again(in same directory) on Sunday/Monday and got the error reported. (I might have recompiled from source again)

if I compiled from source on Friday would that have been a Witti node? I did use the same directory when executing

Thanks
